usmqe/gluster/gluster.py

```diff
@@ -0,0 +1,183 @@
+        """
+        Run command on all gluster nodes
+        """
+        last_error = None
+        output = None
+        if nodes is None:
+            nodes = usmqe.inventory.role2hosts("gluster")
+        if not executor:
+            executor = self.cmd
+        for node in nodes:
+            try:
+                output = executor.run(node, command)
+            except GlusterCommandErrorException as err:
+                last_error = err
+                continue
+            break
+        else:
+            if last_error:
+                raise GlusterCommandErrorException(
+                    "Problem with gluster command '%s'.\n"
+                    "Last command rcode: %s, stdout: %s, stderr: %s" %
+                    (last_error.cmd, last_error.rcode,
+                     last_error.stdout, last_error.stderr))
+            else:
+                raise GlusterCommandErrorException(
+                    "Problem with gluster command '%s'.\n"
+                    "Possible problem is no gluster-node (gluster_node list: %s)" %
+                    (command, usmqe.inventory.role2hosts("gluster")))
+
+        if parse_output:
+            output = parse_output(output)
+        return output
+
+    def get_volume_names(self):
+        """
+        Returns name(s) of volume.
+        TODO specify order or some search if more volumes
+        """
+        vol_name = self.run_on_node(command="volume info").findtext(
+            "./volInfo/volumes/volume/name")
+        LOGGER.debug("Volume_name: %s" % vol_name)
+        return vol_name
+
+    def find_volume_name(self, name, expected=True):
+        """
+        Check if there is volume with given name.
+        """
+        volume_name = self.get_volume_names()
+        found = False
+        # TODO test what vol_name looks like when there are more volumes name
+        if isinstance(volume_name, list):
+            for item in volume_name:
+                # TODO use pytest.check in if?
+                if item == name:
+                    found = True
+        else:
+            if volume_name == name:
+                found = True
+        if expected:
+            pytest.check(found)
+        else:
+            pytest.check(not found)
+        return found
+
+    # TODO do it universal with name checking
+    def check_status(self, name, status="Started"):
+        """
+        Check if volume status corresponds with specified status.
+        """
+        real_status = self.run_on_node(command="volume info").findtext(
+            "./volInfo/volumes/volume/statusStr")
+        LOGGER.debug("Volume_status: %s" % real_status)
+        pytest.check(status == real_status)
+
+
+class GlusterVolume(GlusterCommon):
+    """
+    Class representing gluster cluster.
+    """
+
+    def __init__(self, cluster):
+        """
+        Initialize GlusterCluster object.
+
+        Args:
+            cluster: cluster name
+        """
+        super(GlusterCommon, self).__init__(cluster)
```

Comment:
maybe I am wrong but I think that following line is missing
self.cmd = GlusterVolumeCommand()